Linux main

KCkingcollin opened the pull request:
stable build of linux auto compiling scripts

KCkingcollin committed:
Create Linux-Auto-Compile-Full-Patch.sh

supper simple script to compile and run the ussedp app in linux

KCkingcollin committed:
Rename KCs-Linux-Fork/Linux-Auto-Compile-Full-Patch.sh to KCs-Linux-Scripts/Linux-Auto-Compile-Full-Patch.sh

KCkingcollin committed:
Create Linux-Auto-Compile-Best-Of-Both-Worlds.sh

a supper simple script for downloading and compiling ussedp on linux

KCkingcollin committed:
Create ReadMe.md

mmm ReadMe baby...

KCkingcollin committed:
Update Linux-Auto-Compile-Best-Of-Both-Worlds.sh

did a bit of an oof

KCkingcollin committed:
Update Linux-Auto-Compile-Full-Patch.sh

did a bit of an oof

KCkingcollin committed:
Update Linux-Auto-Compile-Full-Patch.sh

I Fucked it up massively so here is the fix

KCkingcollin committed:
Update Linux-Auto-Compile-Best-Of-Both-Worlds.sh

I Fucked it up massively so here is the fix

KCkingcollin committed:
Update Linux-Auto-Compile-Best-Of-Both-Worlds.sh

KCkingcollin committed:
Update Linux-Auto-Compile-Full-Patch.sh

CrezyDud committed:
CLEEEEEN YES

CrezyDud committed:
CLEEEEEN(R) YEEEYYS

CrezyDud committed:
cleanse ghostis

CrezyDud committed:
Merge remote-tracking branch 'upstream/patch-1' into patch-1

CrezyDud committed:
usage of `whoami`

CrezyDud committed:
more usage of `whoami`

I'm not really sure if `-d` and `-f` needs it, but i've been told it does

CrezyDud committed:
build as release

CrezyDud committed:
build as release

KCkingcollin committed:
Merge pull request #1 from CrezyDud/patch-1

Save poor people

KCkingcollin committed:
Update Linux-Auto-Compile-Full-Patch.sh

KCkingcollin committed:
Update Linux-Auto-Compile-Best-Of-Both-Worlds.sh

KCkingcollin committed:
Update Linux-Auto-Compile-Best-Of-Both-Worlds.sh

just added the google drive link wasn't easy but its fast and reliable tested the command in bash but haven't tested the script yet going to a quick review and do a test

KCkingcollin committed:
Update Linux-Auto-Compile-Full-Patch.sh

just added the google drive link wasn't easy but its fast and reliable tested the command in bash but haven't tested the script yet going to a quick review and do a test

KCkingcollin committed:
Update Linux-Auto-Compile-Best-Of-Both-Worlds.sh

tested and confirmed working

KCkingcollin committed:
Update Linux-Auto-Compile-Full-Patch.sh

not tested yet but should work the same as the other one

KCkingcollin committed:
Update Linux-Auto-Compile-Best-Of-Both-Worlds.sh

missed some code in line 9

KCkingcollin committed:
Update Linux-Auto-Compile-Full-Patch.sh

missed some code in line 9

KCkingcollin committed:
this should work

KCkingcollin committed:
this should work

KCkingcollin committed:
just some touch ups

KCkingcollin committed:
fixed lines 26 and 28

KCkingcollin committed:
fist atempt at using the nexus api to download

KCkingcollin committed:
changed the built path and added a password checker to save users some headache

KCkingcollin committed:
removed the comments so no one thinks the google drive link is active

KCkingcollin committed:
this should make testing a **LOT** easier

KCkingcollin committed:
in the name of simplicity i added gui for the password in case someone wants to open it from outside the terminal

KCkingcollin commented:
sorry didn't realize changing the branch name was going to cancel the pull request [my last comment](https://github.com/wabbajack-tools/ussedp/pull/9#issuecomment-980500082) on old pull

KCkingcollin committed:
Delete DTOs directory

im not sure we need this file for the script...

KCkingcollin committed:
Delete .github/workflows directory

im not sure we need this file for the script...

KCkingcollin committed:
Delete MakeSelfContained directory

im not sure we need this file for the script...

KCkingcollin committed:
Delete PatchBuilder directory

im not sure we need this file for the script...

KCkingcollin committed:
Delete Patcher directory

im not sure we need this file for the script...

KCkingcollin committed:
Delete .gitignore

im not sure we need this file for the script...

KCkingcollin committed:
Delete README.md

im not sure we need this file for the script...

KCkingcollin committed:
Delete buildAll.bat

im not sure we need this file for the script...

KCkingcollin committed:
Delete unex.yml

im not sure we need this file for the script...

KCkingcollin committed:
Delete ussedp.sln

im not sure we need this file for the script...

KCkingcollin committed:
Create README.md

KCkingcollin commented:
just realized that there was no need for those other files to be in this branch

KCkingcollin committed:
Delete README.md

KCkingcollin committed:
Rename KCs-Linux-Scripts/ReadMe.md to KCs-README.md

KCkingcollin committed:
Rename KCs-Linux-Scripts/Linux-Auto-Compile-Full-Patch.sh to Linux-Auto-Compile-Full-Patch.sh

KCkingcollin committed:
Rename KCs-Linux-Scripts/Linux-Auto-Compile-Best-Of-Both-Worlds.sh to Linux-Auto-Compile-Best-Of-Both-Worlds.sh

KCkingcollin commented:
did some **serious** house cleaning in [KCkingcolln/ussedp](https://github.com/KCkingcollin/ussedp) as i plan to continue supporting it so i got rid of any unneeded files

KCkingcollin committed:
Rename KCs-README.md to README-KC.md

KCkingcollin committed:
Update and rename README-KC.md to README.md

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update README.md

KCkingcollin reviewed (commented):
if merged the deleted files wouldn't be a problem right?...
or do they need to be in the repo to merge correctly?

KCkingcollin commented:
Shit I think I did a oof, maybe I should just keep the repo separate as a fork🤔?...

CrezyDud commented:
you still need to ass zenity as dependency

CrezyDud commented:
why did you delete all the source files, this would make merging this remove all the files

CrezyDud commented:
```
$ sudo -S
sudo: invalid option -- 'S'
usage: doas [-Lns] [-C config] [-u user] command [args]
```
-S wont work with opendoas-sudo

KCkingcollin commented:
> ```
> $ sudo -S
> sudo: invalid option -- 'S'
> usage: doas [-Lns] [-C config] [-u user] command [args]
> ```
> -S wont work with opendoas-sudo

Damn ok I'll have to find another way to sudo, I think I was onto something yesterday I'll try it again

KCkingcollin commented:
> why did you delete all the source files, this would make merging this remove all the files

I wasn't sure that was the case I was hoping I could just add the script files but I see that's not possible

btw if this branch is merged does it still exist separately? 

If not it might be more beneficial to leave it as a branch, as the source files are pulled via `get clone` anyway... 

However if the branch is persistent beyond the merge then I guess I'll and the source back so it can be merged...

KCkingcollin commented:
> you still need to add zenity as dependency

Check the [README.md](https://github.com/KCkingcollin/ussedp/blob/linux-main/README.md) I've added the dependency but I think most distros should have it pre installed

KCkingcollin commented:
> > why did you delete all the source files, this would make merging this remove all the files
> 
> I wasn't sure that was the case I was hoping I could just add the script files but I see that's not possible
> 
> btw if this fork is merged does it still exist separately? 
> 
> If not it might be more beneficial to leave it as a fork, as the source files are pulled via `get clone` anyway... 
> 
> However if the branch is persistent beyond the merge then I guess I'll add the source back so it can be merged...

Ok I answered my own question with Google I'll get the source back in the fork pronto...

KCkingcollin committed:
sorry im still new here dint know i needed these

KCkingcollin committed:
Merge branch 'linux-main' of https://github.com/KCkingcollin/ussedp into linux-main

KCkingcollin committed:
idk why but this file wasnt in my trash bin

KCkingcollin committed:
and this file

KCkingcollin committed:
found out why git didnt like it

KCkingcollin committed:
oops

KCkingcollin committed:
hmm

KCkingcollin committed:
Create release.yaml

KCkingcollin commented:
ok i got all the source back in the fork i just need to fix the `sudo -S` problem now

KCkingcollin commented:
@CrezyDud could i ask why your using [opendoas-sudo](https://github.com/Duncaen/OpenDoas)? i dont really understand the benefit as it seems outdated and unsafe defiantly for an arch distro... 

ill try to fix the `sudo -S` issue regardless to keep the script cross distro but as far as im aware, all most every other distro comes with the `sudo -S` option i think this is specific to [opendoas-sudo](https://github.com/Duncaen/OpenDoas)

KCkingcollin committed:
Removed sudo -S and added UI for script choice

KCkingcollin commented:
Ok @CrezyDud surely this should fix our problem...
Test it when you get a chance plz

btw use the gui script or run one of the other scripts with root

KCkingcollin committed:
updated the download link and updated the README

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update README.md

KCkingcollin commented:
@halgari any idea why the link i generated didn't last?

I tried adding 100000000000... to the "Expiry of the key" hopefully this time it'll stick...

KCkingcollin committed:
added progress windows and merged the scripts

KCkingcollin committed:
Merge branch 'linux-main' of https://github.com/KCkingcollin/ussedp into linux-main

KCkingcollin committed:
Update README.md

KCkingcollin commented:
BTW if anyone wants to test the scripts i would recommend re-cloning\ downloading the fork as a lot has been changed\ added

KCkingcollin commented:
BTW if anyone wants to test the scripts i would recommend re-cloning\ downloading the fork as a lot has been changed\ added

KCkingcollin commented:
ok... i guess ill test it on a VM...

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update README.md

KCkingcollin committed:
removed gksu dependency

KCkingcollin committed:
Merge branch 'linux-main' of https://github.com/KCkingcollin/ussedp into linux-main

KCkingcollin committed:
updated dependencys

KCkingcollin commented:
ok after installing arch i now realize how out of date `gksu` is, so i got ride of it

now using `sudo -A` i already know this doesn't work with `doas` but for now default `sudo` is required

KCkingcollin committed:
Delete patches-download-link

no longer needed

KCkingcollin committed:
Update Linux-Auto-Compile-Script.sh

nexus api links are not a good solution as they expire every day it seems

replacing with google drive link again until i can find some other way to download the patch files directly from @halgari

mastercoms commented:
why does it download from google docs

KCkingcollin commented:
> why does it download from google docs

Because it's currently not possible for me to use the Nexus link as the script uses `wget` which I can't get to work the only other alternative was to use a link gen that only lasted a day

KCkingcollin committed:
fix for doas users and auto install dependencies

KCkingcollin commented:
added support for `doas` users i think, i haven't been able to fully test it yet, in theory it should work though
also added auto dependency check and will prompt the user to install missing dependencies

KCkingcollin committed:
forgot to add zenity width for a prompt

KCkingcollin commented:
will be fixing the arch problem in [linux-main #6](https://github.com/KCkingcollin/ussedp/issues/6) soon i just wanted to help out the people who use `doas` fist and also make it just a bit more convenient

KCkingcollin commented:
hey @CrezyDud if you still keeping up with my fork id greatly appreciate it if you could test the `doas` support of the new release
im not very familiar with `doas` so i could only really make assumptions based off of what you have said and what google has said about it...

KCkingcollin committed:
Update README.md

KCkingcollin commented:
arch based distros confirmed to be working doing the last bit of testing on kubuntu and that should wrap this project up...

KCkingcollin committed:
Update Linux-Auto-Compile-Script.sh

KCkingcollin committed:
Update Linux-Auto-Compile-Script.sh

KCkingcollin committed:
Update README.md

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

Ubuntu needed the specific "6.0" package so i fixed it

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

ok now its fixed... I hope... also removed apk seeing as no one uses that anymore...

KCkingcollin committed:
Update README.md

KCkingcollin commented:
Ubuntu based distros confirmed working! this should mean most if not all debian based distros should work with this script

onto fedora....

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

did some adjusting for fedora

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

idk why but fedora dose not have the 6.0 sdk in the repos yet?...

KCkingcollin committed:
Update README.md

i think the scrip will work with 5.0 so i changed the dependency

KCkingcollin committed:
Update README.md

nope nevermined

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

added a quick version check for dotnet after finding out that 5.0 dose not work with the script

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

had a problem with wget in fedora

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

forgot to `sudo` something and put an `rm` in the wrong place

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

lol forgot to chmod the script

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

didnt need sudo for dotnet script

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

that should fix things up for fedora

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

almost forgot about removing the old dotnet executable

KCkingcollin commented:
fedora support not currently ready idk why but fedora 7z dose not work as expected and i need to probably detect fedora and do some special stuff in the script

monyarm commented:
Many Gentoo users use doas instead of sudo, and doas doesn't have a -A argument, in fact it doesn't have askpass.

KCkingcollin commented:
> Many Gentoo users use doas instead of sudo, and doas doesn't have a -A argument, in fact it doesn't have askpass.

I'm aware of this fact which is why their should be a failsafe in place simply run the gui script in a terminal like normal and you should get a normal password prompt from `doas` in the event that askpass fails to return a correct password, but i have not been able to test this yet as i currently do not use `doas` nor do i have a vm setup with a distro that comes with it...

KCkingcollin committed:
Update Linux-Auto-Compile-Script.sh

fixed "FullPatcher" option not working right and updated the patch files

KCkingcollin committed:
Update USSEDP-GUI-Linux.sh

updated for patch files version 1.5.3

KCkingcollin commented:
lol sorry any one who wanted to use the "FullPatcher" option till now, i just found out when updating the scripts patch files to 1.5.3 that i forgot to add an important variable to getting the option system to work up until now the script has been only doing the "best of both worlds" option regardless of the users input sorry about that i thought for sure i replaced all the old code with the new `$choice` variable but i missed the most important part in line #39 of the linux-auto-compile-script.sh
